### PR TITLE
Fix bug which would keep L2 component of gradient and Hessian for intercept term

### PR DESCRIPTION
**Pull request by ashelkovnykov**

### CODE REVIEW

File: photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala [hunk 1]
```diff
@@ -196,25 +208,52 @@ trait L2RegularizationTwiceDiff extends TwiceDiffFunction with L2RegularizationD
...
+
+      case None =>
+    }
+
+    l2HessianVector
+  }
 
   /**
    * Compute the Hessian diagonal of the L2 regularization term.
    *
```
Comment (yunboouyang) [inline photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala]: Need to add `@param` for `DIMENSION`

File: photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala [hunk 2]
```diff
@@ -174,7 +186,7 @@ trait L2RegularizationTwiceDiff extends TwiceDiffFunction with L2RegularizationD
       coefficients: Coefficients,
       multiplyVector: Coefficients,
       normalizationContext: BroadcastWrapper[NormalizationContext]): Vector[Double] =
-    super.hessianVector(input, coefficients, multiplyVector, normalizationContext) +
+    super.hessianVector(input, coefficients, multiplyVector, normalizationContext) +:+
```
Comment (yunboouyang) [inline photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala]: Why do we change to `+:+`? `+` supports element-wise addition

File: photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala [hunk 3]
```diff
@@ -196,25 +208,52 @@ trait L2RegularizationTwiceDiff extends TwiceDiffFunction with L2RegularizationD
...
+
+      case None =>
+    }
+
+    l2HessianDiagonal
+  }
+
+  /**
+   * Compute the Hessian matrix of the L2 regularization term.
+   *
```
Comment (cmjiang) [inline photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala]: Miss a param here too.

File: photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala (hunk 2 above)
Comment (ashelkovnykov) [reply photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala]: Trying to remember - I think my train of thought was that `+` would be for scalar values, and `+:+` would be for vector values. I think I was trying to make it clear what type of values are being summed by looking at which symbol is being used.

**Pull request merged**